### PR TITLE
Update Thanos version to master

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR updates Thanos dependency to (214ff4480e93).

Work in progress
- [x] integrate changes in compactor: delayed deletes, separate blocks cleanup
- [x] expose postings compression in the cache
- [ ] update documentation with new flags
- [x] verify metric schanges
- [x] expose memcache client object size option

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/storage/tsdb/config.go [hunk 1]
```diff
@@ -156,17 +156,19 @@ func (cfg *Config) Validate() error {
...
+	SyncDir                  string           `yaml:"sync_dir"`
+	SyncInterval             time.Duration    `yaml:"sync_interval"`
+	MaxChunkPoolBytes        uint64           `yaml:"max_chunk_pool_bytes"`
+	MaxSampleCount           uint64           `yaml:"max_sample_count"`
+	MaxConcurrent            int              `yaml:"max_concurrent"`
+	TenantSyncConcurrency    int              `yaml:"tenant_sync_concurrency"`
+	BlockSyncConcurrency     int              `yaml:"block_sync_concurrency"`
+	MetaSyncConcurrency      int              `yaml:"meta_sync_concurrency"`
+	BinaryIndexHeader        bool             `yaml:"binary_index_header_enabled"`
+	PostingsCompressions     bool             `yaml:"postings_compression_enabled"`
```
Comment (pracucci) [inline pkg/storage/tsdb/config.go]: This is a duplicate of the setting in the index cache and should be removed.

File: pkg/storage/tsdb/index_cache.go [hunk 2]
```diff
@@ -48,6 +50,7 @@ func (cfg *IndexCacheConfig) RegisterFlags(f *flag.FlagSet) {
 
 func (cfg *IndexCacheConfig) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.StringVar(&cfg.Backend, prefix+"backend", IndexCacheBackendDefault, fmt.Sprintf("The index cache backend type. Supported values: %s.", strings.Join(supportedIndexCacheBackends, ", ")))
+	f.BoolVar(&cfg.PostingsCompression, "postings-compression-enabled", false, "Compress postings before storing them to postings cache.")
```
Comment (pracucci) [inline pkg/storage/tsdb/index_cache.go]: `prefix` is missing here

File: pkg/compactor/compactor.go [hunk 3]
```diff
@@ -149,11 +158,24 @@ func newCompactor(
 			Name: "cortex_compactor_runs_failed_total",
 			Help: "Total number of compaction runs failed.",
 		}),
+
+		blocksCleaned: prometheus.NewCounter(prometheus.CounterOpts{
```
Comment (pracucci) [inline pkg/compactor/compactor.go]: May we use `promauto.With(reg)` and get rid of `registerer.MustRegister()` please?

File: pkg/storage/tsdb/config.go (hunk 1 above)
Comment (pstibrany) [reply pkg/storage/tsdb/config.go]: Thanks for catching this! It's a leftover before I decided to move it to IndexCacheConfig.

**Pull request merged**